src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/TreeMaker.java

```diff
@@ -257,6 +257,37 @@ private static boolean isAnonymousStruct(Declaration declaration) {
                 return null;
             }
         }
+        Type.Function funcType = null;
+        boolean isFuncPtrType = false;
+        if (canonicalType instanceof Type.Function) {
```

Comment:
I believe `OutputFactory::getAsFunctionPointer` as code that is similar to this (given a type, return it as a function type - following typedefs if necessary). Maybe we should move this into some util function and share?

Reply:
That part of the code assumes all Types are properly resolved. In TreeMaker, we're in the middle TypeMaker's constructor (resolution happens later).

Reply:
refactored pointer type check to Utils.java.